src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -76,6 +77,21 @@ public sealed class VsSolutionRestoreService : IVsSolutionRestoreService
 
         public Task<bool> CurrentRestoreOperation => _restoreWorker.CurrentRestoreOperation;
 
+        public Task<bool> NominateProjectAsync(string projectUniqueName, CancellationToken token)
+        {
+            if (string.IsNullOrEmpty(projectUniqueName))
```

Comment:
We recently started using it. Already use it for Argument null check, it's more convenient and simple without worrying about right error message as well.